app/controllers/user.js

```diff
@@ -87,3 +88,38 @@ exports.userList = async (req, res, next) => {
     next(err);
   }
 };
+
+exports.singUpAdmins = async (req, res, next) => {
+  const auth = req.headers[AUTHORIZATION];
+
+  const plainText = decoder(auth);
+
+  try {
+    const requester = await User.getUserBy(plainText.email);
+
+    const permissionsErrors = validateAdmin(requester);
+
+    if (!permissionsErrors.valid) throw errors.authorizationError(permissionsErrors.messages);
```

Comment:
A middleware for this has a perfect match here, in that way do you don't need repeat this code everywhere

Reply:
i should do another service  in te middleware filen, one for `list users` and other to validate if is an administrator user for consume `/admin/users`

Reply:
In this case the only middleware that you need is for check if user is or not a admin